JsonParser.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include <iostream>
+#include "Unit.h"
+#include <string>
+#include <map>
+#include <variant>
+#include <fstream>
+#include <algorithm>
+
+class JsonParser{
+private:
+public:
+  static std::map<std::string, std::string> parseJSON(std::istream& data);
+  static std::map<std::string, std::string> parseJSON(const std::string& data);
```

Comment:
Hiányzik a 3. fv.: "Egy jsonparser branchen keszuljon el egy altalanosabb JSON parser, mely 3 kulonbozo formaban kaphat JSON bemenetet: fajlnev, istream, string."

Reply:
`std::ifstream test(data);
 if(test){...`
Ez a kódrészlet ellenőrzi, hogy a bemeneti string az egy érvényes fájlnév-e. Ily módon lehetséges fájlnevet, istreamet és stringet megadva is parse-olni. A feladat kiírás szerint nem kötelező 3 függvényt létrehozni, a lényeg, hogy mindhárom bemenetre egy jól parsolt kimenetet adjon az osztály, amit szerintem így meg tud valósítani.

Reply:
Azért lenne tisztább a helyzet 3 fv.-nyel, mert könnyebben értelmezhető üzeneteket lehet adni hiba esetén. Könnyebb elválasztani, hogy a fájl nem létezik, mert pl. rossz nevet adtak meg, vagy az nem is fájlnév, hanem hibásan formázott json tartalom. És nem okozna ez sok plusz munkát, mert az egyik működését nagyon könnyű visszavezetni a másikra.